src/main/java/com/github/tomakehurst/wiremock/extension/requestfilter/RequestWrapper.java

```diff
@@ -180,21 +179,21 @@ public Set<String> getAllHeaderKeys() {
 
   @Override
   public Map<String, Cookie> getCookies() {
-    ImmutableMap.Builder<String, Cookie> builder = ImmutableMap.builder();
+    Map<String, Cookie> cookieMap = new HashMap<>();
     for (Map.Entry<String, Cookie> entry : delegate.getCookies().entrySet()) {
       Cookie newCookie =
           cookieTransformers.containsKey(entry.getKey())
               ? cookieTransformers.get(entry.getKey()).transform(entry.getValue())
               : entry.getValue();
 
       if (!cookiesToRemove.contains(entry.getKey())) {
-        builder.put(entry.getKey(), newCookie);
+        cookieMap.put(entry.getKey(), newCookie);
       }
     }
 
-    builder.putAll(additionalCookies);
+    cookieMap.putAll(additionalCookies);
 
-    return builder.build();
+    return cookieMap;
```

Comment:
Can we preserve immutability here e.g. by calling Collections.unmodifiableMap()?